File: php/testcode/benchmark_test_00345.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00345(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('message');

    $safe = htmlspecialchars($input, ENT_QUOTES, 'UTF-8');
    $html = '<p>' . $safe . '</p>';

    return BenchmarkResponse::html($html);
}

File: rust/testcode/benchmark_test_00345.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let page: u64 = req.param("page").parse().unwrap_or(1);
    let offset = (page - 1) * 20;
    super::shared::BenchmarkResponse::ok(&format!("Page {} offset {}", page, offset))
}
